From a8dff18f98e3e3b80d4f8ec18b700df96b4d3b92 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Wed, 13 Nov 2019 02:30:57 -0800
Subject: [PATCH 0684/1852] v4l: xilinx: xcsi2rxss: Set default active lanes to
 max

commit 8ce6be368c057e57f652d80c9f3d2da95f30c838 from
https://github.com/Xilinx/linux-xlnx.git

Fix the default active lanes from minimum value 1 to max lanes as
defined in the device tree for ease of use. Usually users had to set the
active lanes control to the max lanes. Now this is not needed.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-csi2rxss.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-csi2rxss.c b/drivers/media/platform/xilinx/xilinx-csi2rxss.c
index 62e7336ec7e5..4368fbcc1319 100644
--- a/drivers/media/platform/xilinx/xilinx-csi2rxss.c
+++ b/drivers/media/platform/xilinx/xilinx-csi2rxss.c
@@ -1968,6 +1968,8 @@ static int xcsi2rxss_probe(struct platform_device *pdev)
 			if (xcsi2rxss->core.enable_active_lanes) {
 				xcsi2rxss_ctrls[i].max =
 					xcsi2rxss->core.max_num_lanes;
+				xcsi2rxss_ctrls[i].def =
+					xcsi2rxss->core.max_num_lanes;
 			} else {
 				/* Don't register control */
 				dev_dbg(xcsi2rxss->core.dev,
-- 
2.31.1

